### PR TITLE
refactor(tracing): move tracing setup into own package and extra tests added

### PR DESCRIPTION
**Pull request by erka**

### CODE REVIEW

File: internal/tracing/tracing.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/exporters/zipkin"
+	"go.opentelemetry.io/otel/sdk/resource"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+// newResourse constructs a trace resource with Flipt-specific attributes.
+// It incorporates schema URL, service name, service version, and OTPL environment data
+func newResourse(ctx context.Context, fliptVersion string) (*resource.Resource, error) {
```
Comment (markphelps) [inline internal/tracing/tracing.go:23]:
```suggestion
func newResource(ctx context.Context, fliptVersion string) (*resource.Resource, error) {
```

File: internal/tracing/tracing.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/exporters/zipkin"
+	"go.opentelemetry.io/otel/sdk/resource"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+// newResourse constructs a trace resource with Flipt-specific attributes.
+// It incorporates schema URL, service name, service version, and OTPL environment data
```
Comment (markphelps) [inline internal/tracing/tracing.go:22]:
```suggestion
// It incorporates schema URL, service name, service version, and OTLP environment data
```

File: internal/tracing/tracing_test.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.flipt.io/flipt/internal/config"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+func TestNewResourseDefault(t *testing.T) {
```
Comment (markphelps) [inline internal/tracing/tracing_test.go:15]:
```suggestion
func TestNewResourceDefault(t *testing.T) {
```

**Review by markphelps — COMMENTED** [review]
looks great @erka ! thank you! just a couple spelling suggestions

**Review by yquansah — APPROVED** [review]
Looks good on my end.

**Review by markphelps — APPROVED** [review]

**Pull request merged**